### PR TITLE
AT-8718 Add missing validation message on summary

### PR DESCRIPTION
**Pull request by tomarnold-ccpo**
Added the `entireDuration` field
Added logic to check if specific periods were selected

### CODE REVIEW

File: src/components/ATATCheckboxGroup.vue [hunk 1]
```diff
@@ -422,6 +426,14 @@ export default class ATATCheckboxGroup extends Vue {
 
   public mounted(): void {
     this.setEventListeners();
+   
+    // if validateOnLoad, then validate checkboxes immediately
+    if (this.validateOnLoad){
+      this.validateCheckboxesNow = true;
+      setTimeout(()=>{
+        this.setErrorMessage();
+      }, 1000)
```
Comment (ericyoungquist-ccpo) [inline src/components/ATATCheckboxGroup.vue:435]: does timeout of `0` also work?

File: src/components/DOW/AnticipatedDurationandUsage.vue [hunk 2]
```diff
@@ -129,9 +128,19 @@ export default class AnticipatedDurationandUsage extends Vue {
...
+        : [];
+    }
+
+    if (newVal === "NO"){
+      this.periodCheckboxRules.push(
+        this.$validators.required('Please select at least one base or option period.')
+      )
+    } else {
+      this.periodCheckboxRules = [];
+    }
```
Comment (ericyoungquist-ccpo) [inline src/components/DOW/AnticipatedDurationandUsage.vue:143]: could do the above in a ternary with fewer lines of code:
```
this.periodCheckboxRules = newVal === "NO" 
    ? [this.$validators.required('Please select at least one base or option period.')]
    : [];
```

File: src/steps/05-PerformanceRequirements/DOW/OtherOfferingSummary.vue [hunk 3]
```diff
@@ -512,28 +521,32 @@ export default class OtherOfferingSummary extends Mixins(SaveOnLeave) {
...
-      }
     }
-
-    requiredFields.forEach((field) => {
-      if (instanceData[field] === "") {
-        isValid = false;
-      }
-    });
-
+    console.log(instanceData)
```
Comment (ericyoungquist-ccpo) [inline src/steps/05-PerformanceRequirements/DOW/OtherOfferingSummary.vue:535]: remove console.log

File: src/components/ATATCheckboxGroup.vue (hunk 1 above)
Comment (tonysumner-ccpo) [reply src/components/ATATCheckboxGroup.vue:435]: no, that was my first try...

File: src/steps/05-PerformanceRequirements/DOW/OtherOfferingSummary.vue [hunk 4]
```diff
@@ -543,7 +556,7 @@ export default class OtherOfferingSummary extends Mixins(SaveOnLeave) {
     this.isCompute = offering === "compute";
     //depreciate since each serviceOfferingGroupId is unique 
     //(ADVISORY_ASSISTANCE | HELP_DESK_SERVICES | DOCUMENTATION_SUPPORT | GENERAL_CLOUD_SUPPORT)
```
Comment (ericyoungquist-ccpo) [inline src/steps/05-PerformanceRequirements/DOW/OtherOfferingSummary.vue:558]: pls remove commented code

File: src/components/ATATCheckboxGroup.vue [hunk 5]
```diff
@@ -397,6 +399,8 @@ export default class ATATCheckboxGroup extends Vue {
   public validateNowChange(): void {
     if (!this.validateCheckboxesNow && this.rules.length) {
       this.validateCheckboxesNow = true;
+    } else if (this.validateOnLoad){
```
[the remark this answers is not in the document]
Comment (tonysumner-ccpo) [reply src/components/ATATCheckboxGroup.vue:402]: done

File: src/components/DOW/AnticipatedDurationandUsage.vue (hunk 2 above)
Comment (tonysumner-ccpo) [reply src/components/DOW/AnticipatedDurationandUsage.vue:143]: done

File: src/steps/05-PerformanceRequirements/DOW/OtherOfferingSummary.vue (hunk 3 above)
Comment (tonysumner-ccpo) [reply src/steps/05-PerformanceRequirements/DOW/OtherOfferingSummary.vue:535]: removed

File: src/steps/05-PerformanceRequirements/DOW/OtherOfferingSummary.vue (hunk 4 above)
Comment (tonysumner-ccpo) [reply src/steps/05-PerformanceRequirements/DOW/OtherOfferingSummary.vue:558]: done.

**Review by tonysumner-ccpo — APPROVED** [review]

**Review by ericyoungquist-ccpo — COMMENTED** [review]
Approved!

**Pull request merged**